packages/core/src/payment/strategies/credit-card-redirect/credit-card-redirect-payment-strategy.ts

```diff
@@ -85,11 +88,15 @@ export default class CreditCardRedirectPaymentStrategy extends CreditCardPayment
                 return Promise.reject(error);
             }
 
-            return new Promise(() => this._formPoster.postForm(error.body.three_ds_result.acs_url, {
-                PaReq: error.body.three_ds_result.payer_auth_request || null,
-                TermUrl: error.body.three_ds_result.callback_url || null,
-                MD: error.body.three_ds_result.merchant_data || null,
-            }));
+            return new Promise(() => {
+                if (isRequestError(error)) {
```

Comment:
Same as above, you can store the `three_ds_result` in a variable instead of doing another check, e.g.:

```ts
const { three_ds_result } = error.body;
```